### DISCUSSION TITLE
uno Multi-platform app cant build

### DISCUSSION THREAD

**Discussion by Ali1Jamal** [unoplatform/uno#7648] [Q&A]
i'm sure everything installed successfully and also maui is working without problems but I wasn't able to use uno because of the problems I'm facing.
the errors : 
1-
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1012	Platform version is not present for one or more target frameworks, even though they have specified a platform: net6.0-macos	UnoApp1.Mobile	F:\Projects\UnoApp1\UnoApp1\UnoApp1.Mobile\UnoApp1.Mobile.csproj	1	
2-
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS2001	Source file 'F:\Projects\UnoApp1\UnoApp1\UnoApp1.UWP\obj\x64\Debug\App.g.i.cs' could not be found.	UnoApp1.UWP	F:\Projects\UnoApp1\UnoApp1\UnoApp1.UWP\CSC	1	Active
3-
Severity	Code	Description	Project	File	Line	Suppression State
Error		The "CompileXaml" task returned false but did not log an error.	UnoApp1.UWP			

I'm also get failed to download some packages when start build :

Build started...
Failed to download package 'Uno.UI.RemoteControl.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui.remotecontrol/4.0.11/uno.ui.remotecontrol.4.0.11.nupkg'.
Cannot access a disposed object.
Object name: 'SslStream'.
Failed to download package 'Microsoft.UI.Xaml.2.6.2' from 'https://api.nuget.org/v3-flatcontainer/microsoft.ui.xaml/2.6.2/microsoft.ui.xaml.2.6.2.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.ui.xaml/2.6.2/microsoft.ui.xaml.2.6.2.nupkg' has timed out after 100000ms.
Failed to download package 'Microsoft.Extensions.Logging.5.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/5.0.0/microsoft.extensions.logging.5.0.0.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/5.0.0/microsoft.extensions.logging.5.0.0.nupkg' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.foundation.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.foundation.logging/index.json' has timed out after 100000ms.
Failed to download package 'Uno.UI.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui/4.0.11/uno.ui.4.0.11.nupkg'.
Cannot access a disposed object.
Object name: 'SslStream'.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.core.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.core.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options.configurationextensions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options.configurationextensions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.runtime.skia.gtk/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.runtime.skia.gtk/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.configuration/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.configuration/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.buffers/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.buffers/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.text.json/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.text.json/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.memory/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.memory/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/commonservicelocator/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/commonservicelocator/index.json' has timed out after 100000ms.
Failed to download package 'Uno.UI.Skia.Wpf.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/4.0.11/uno.ui.skia.wpf.4.0.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/4.0.11/uno.ui.skia.wpf.4.0.11.nupkg' has timed out after 100000ms.
Failed to download package 'Microsoft.NETCore.UniversalWindowsPlatform.6.2.11' from 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.platforms/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.platforms/index.json' has timed out after 100000ms.
Failed to download package 'Microsoft.Extensions.Logging.5.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/5.0.0/microsoft.extensions.logging.5.0.0.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/5.0.0/microsoft.extensions.logging.5.0.0.nupkg' has timed out after 100000ms.
Failed to download package 'Uno.UI.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui/4.0.11/uno.ui.4.0.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/4.0.11/uno.ui.4.0.11.nupkg' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.foundation.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.foundation.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.core.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.core.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.runtime.skia.gtk/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.runtime.skia.gtk/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.configuration/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.configuration/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options.configurationextensions/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options.configurationextensions/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.buffers/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.buffers/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.text.json/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.text.json/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.memory/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/system.memory/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/commonservicelocator/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/commonservicelocator/index.json' has timed out after 100000ms.
Failed to download package 'Uno.UI.Skia.Wpf.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/4.0.11/uno.ui.skia.wpf.4.0.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/4.0.11/uno.ui.skia.wpf.4.0.11.nupkg' has timed out after 100000ms.
Failed to download package 'Microsoft.NETCore.UniversalWindowsPlatform.6.2.11' from 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.platforms/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.platforms/index.json' has timed out after 100000ms.
Failed to download package 'Uno.UI.4.0.11' from 'https://api.nuget.org/v3-flatcontainer/uno.ui/4.0.11/uno.ui.4.0.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/4.0.11/uno.ui.4.0.11.nupkg' has timed out after 100000ms.
Failed to download package 'Microsoft.NETCore.UniversalWindowsPlatform.6.2.11' from 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/6.2.11/microsoft.netcore.universalwindowsplatform.6.2.11.nupkg' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/uno.ui.skia.wpf/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json'.
The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json' has timed out after 100000ms.
NuGet package restore failed. Please see Error List window for detailed warnings and errors.
Error occurred while restoring NuGet packages: Failed to retrieve information about 'Microsoft.NETCore.UniversalWindowsPlatform' from remote source 'https://api.nuget.org/v3-flatcontainer/microsoft.netcore.universalwindowsplatform/index.json'.
1>------ Build started: Project: UnoApp1.UWP, Configuration: Debug x64 ------
1>F:\Projects\UnoApp1\UnoApp1\UnoApp1.UWP\UnoApp1.UWP.csproj : XamlCompiler error WMC1006: Cannot resolve Assembly or Windows Metadata file 'Type universe cannot resolve assembly: System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a.'
1>CSC : error CS2001: Source file 'F:\Projects\UnoApp1\UnoApp1\UnoApp1.UWP\obj\x64\Debug\App.g.i.cs' could not be found.
1>C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\XamlCompiler\Microsoft.Windows.UI.Xaml.Common.targets(482,5): error MSB4181: The "CompileXaml" task returned false but did not log an error.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========

**Comment by jeromelaban** [comment]
Thanks for the report. This seems like nuget restore issue. Until a nuget restore can be completed properly, we cannot diagnose further.

Try deleting the bin and obj folders, or clearing your nuget cache.

**Reply by Ali1Jamal** [reply]
@jeromelaban 
I cleaned bin , obj and clean solution. I got a message that I have to install workload macos and I installed it successfully.
..but still getting errors installing nuget packages.
is there are any problem in my package sources ?
![Capture](https://user-images.githubusercontent.com/40013555/145914951-f6e318df-0f36-4726-9aec-14b6742841a9.PNG)

**Reply by jeromelaban** [reply]
I do not see anything specific, but until nuget restores properly, there's nothing Uno can do... Try building using the command line (using `msbuild /r youproject.csproj`) maybe  it'll use a different path.

**Reply by ToddGlodek** [reply]
Just upgraded to VisualStudio 2022 Preview (Version 17.1.0 Preview 3.0) -  Observed described build errors immediately after upgrade (still targeting 10.0 Build 19041).  The problem presented in a Uno Multi-platform library.  Using UnoPlatform 4.0.11.9

Ran the same project using VisualStudio 2022 (Version 17.0.5 - Not Preview)  worked fine.

Eliminated all the target frameworks except for uap10.0.19041 -- then it worked fine for UWP.   Will restore the other target frameworks  and report back shortly.

**Reply by ToddGlodek** [reply]
UPDATE :  switching to  10.0.19041 and removing only the 'macos' target frameworks  allowed it to work using (Version 17.1.0 Preview 3.0)

**Reply by jeromelaban** [reply]
Thanks for the update. W.r.t Preview 2 and 3, we're giving some updates here: https://github.com/unoplatform/uno/discussions/7833

**Comment by ToddGlodek** [comment]
Thanks for the guidance.  

- Updated to uno.check --version 1.1.0-dev.9
- Re-ran uno.check 

However, still not compiling with  net6.0-macos target frameworks using (Version 17.1.0 Preview 3.0 )

**Comment by ghost** [comment]
This worked for me https://stackoverflow.com/questions/69890500/dotnet-restore-gives-301-error-for-a-nuget-package
